x-pack/legacy/plugins/maps/public/layers/styles/vector/properties/dynamic_size_property.js

```diff
@@ -121,25 +121,28 @@ export class DynamicSizeProperty extends DynamicStyleProperty {
   }
 
   getMbSizeExpression() {
-    if (this._isSizeDynamicConfigComplete(this._options)) {
-      return this._getMbDataDrivenSize({
-        targetName: this.getComputedFieldName(),
-        minSize: this._options.minSize,
-        maxSize: this._options.maxSize,
-      });
+    if (!this._isSizeDynamicConfigComplete(this._options)) {
+      return null;
     }
-    return null;
+
+    return this._getMbDataDrivenSize({
+      targetName: this.getComputedFieldName(),
+      minSize: this._options.minSize,
+      maxSize: this._options.maxSize,
+    });
   }
 
   _getMbDataDrivenSize({ targetName, minSize, maxSize }) {
+    const rangeFieldMeta = this.getFieldMeta();
```

Comment:
How about passing rangeFieldMeta into this method? There is a lot of mental load to know that `_isSizeDynamicConfigComplete` checked to ensure rangeFieldMeta exists and that context is lost in this method. Then the check in `getMbSizeExpression` would look like `!this._isSizeDynamicConfigComplete(this._options) || !rangeFieldMeta`